mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -218,24 +222,44 @@ class SetpointRawPlugin : public plugin::PluginBase,
 	{
 		Eigen::Quaterniond desired_orientation;
 		Eigen::Vector3d baselink_angular_rate;
-
+		Eigen::Vector3d body_rate;
+		double thrust;
+		// Set Thrust scaling in px4_config.yaml, setpoint_raw block.
+		if (!sp_nh.getParam("thrust_scaling", thrust_scaling) && req->thrust != 0.0) {
+			ignore_thrust_message_ = true;
+		}
+
+		if (ignore_thrust_message_) {
+			ROS_FATAL("Recieved thrust, but ignore_thrust_message_ is true: "
+				"the most likely cause of this is a failure to specify the thrust_scaling parameters "
+				"on px4/apm_config.yaml. Thrust will be set to 0.");
+			thrust = 0.0;
+			return;
```

Comment:
I'm not sure that you want just drop message (because you return before send).

Reply:
Yes, here I just want to print the warning and return without sending anything - for safety. If the user didnt specify the thrust scaling, then we don't want to do anything.

Reply:
> Thrust will be set to 0.

Is confusing, since you just drop whole message.

Reply:
Yep, I changed that comment and removed thrust zero-ing (it's not needed since we immediately return).